src/main/scala/com/microsoft/hyperspace/index/rules/RuleUtils.scala

```diff
@@ -118,9 +130,26 @@ object RuleUtils {
           IndexLogEntryTags.HYBRIDSCAN_REQUIRED,
           !(commonCnt == entry.sourceFileInfoSet.size && commonCnt == inputSourceFiles.size))
       }
+      entry.setTagValue(plan, IndexLogEntryTags.IS_HYBRIDSCAN_CANDIDATE, isCandidate)
       isCandidate
     }
 
+    def prepareHybridScanCandidateSelection(
```

Comment:
Not related to this PR, but we need to think about refactoring RuleUtils.scala. This is already two levels deep (`getCandidateIndexes` -> `isHybridScanCandidate` -> `prepareHybridScanCandidateSelection`).